service-front/module/Application/src/Controller/DonorFlowController.php

```diff
@@ -156,29 +174,32 @@ public function donorLpaCheckAction(): ViewModel
             // not yet implemented
 //          $response =  $this->opgApiService->saveLpaRefsToIdCheck();
 
-            switch ($detailsData['idMethod']) {
-                case 'pn':
-                    $this->redirect()
-                        ->toRoute("root/passport_number", ['uuid' => $uuid]);
-                    break;
-
-                case 'dln':
-                    $this->redirect()
-                        ->toRoute("root/driving_licence_number", ['uuid' => $uuid]);
-                    break;
-
-                case 'nin':
-                    $this->redirect()
-                        ->toRoute("root/national_insurance_number", ['uuid' => $uuid]);
-                    break;
-
-                case 'po':
-                    $this->redirect()
-                        ->toRoute("root/post_office_documents", ['uuid' => $uuid]);
-                    break;
-
-                default:
-                    break;
+            /**
+             * @psalm-suppress PossiblyUndefinedArrayOffset
+             */
+            if ($detailsData['idMethodIncludingNation']['id_route'] == 'POST_OFFICE') {
+                $this->redirect()
+                    ->toRoute("root/post_office_documents", ['uuid' => $uuid]);
+            } else {
+                switch ($detailsData['idMethodIncludingNation']['id_method']) {
+                    case 'PASSPORT':
```

Comment:
Could use enum values here rather than strings for more clarity:
```suggestion
                    case IdMethod::PASSPORT:
```